modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/execution/executor/MethodExecutorGenerator.java

```diff
@@ -0,0 +1,265 @@
+import net.bytebuddy.dynamic.DynamicType.Unloaded;
+import net.bytebuddy.dynamic.scaffold.InstrumentedType;
+import net.bytebuddy.implementation.Implementation;
+import net.bytebuddy.implementation.MethodCall;
+import net.bytebuddy.implementation.MethodCall.ArgumentLoader;
+import net.bytebuddy.implementation.bytecode.ByteCodeAppender;
+import net.bytebuddy.implementation.bytecode.StackManipulation;
+import net.bytebuddy.implementation.bytecode.assign.Assigner;
+import net.bytebuddy.implementation.bytecode.member.FieldAccess;
+import net.bytebuddy.implementation.bytecode.member.MethodInvocation;
+import net.bytebuddy.implementation.bytecode.member.MethodVariableAccess;
+
+/**
+ * Uses bytecode manipulation to dynamically generate {@link MethodExecutor} classes that invoke a given method.
+ *
+ * @since 4.3.0
+ */
+public class MethodExecutorGenerator {
+
+  private static final String TARGET_INSTANCE_FIELD_NAME = "__targetInstance";
+  private final Map<String, GeneratedClass<MethodExecutor>> executorClasses = new HashMap<>();
+  private final int instanceId = identityHashCode(this);
+
+  /**
+   * Creates a {@link GeneratedInstance} pointing to a dynamic {@link MethodExecutor} that executes the given {@code method}.
+   * <p>
+   * Each invocation to this method will return a new {@link GeneratedInstance} pointing to unique instances. However, all
+   * invocations pointing to the same {@code method} will share the same underlying {@link GeneratedClass}
+   *
+   * @param targetInstance           the instance on which the method is to be executed
+   * @param method                   the method to be invoked
+   * @param argumentResolverDelegate the {@link ArgumentResolverDelegate} that provides the {@link ArgumentResolver resolvers}
+   * @return a {@link GeneratedInstance}
+   * @throws Exception if the instance cannot be generated
+   */
+  public GeneratedInstance<MethodExecutor> generate(Object targetInstance,
+                                                    Method method,
+                                                    ArgumentResolverDelegate argumentResolverDelegate)
+      throws Exception {
+
+    GeneratedClass<MethodExecutor> generatedClass = getExecutorClass(method, targetInstance);
+    List<Object> args = new ArrayList<>();
+    args.add(targetInstance);
+    args.addAll(asList(argumentResolverDelegate.getArgumentResolvers()));
+
+    MethodExecutor instance = (MethodExecutor) generatedClass.getGeneratedClass().getConstructors()[0]
+        .newInstance(args.toArray(new Object[args.size()]));
+
+    return new GeneratedInstance<>(instance, generatedClass);
+  }
+
+  private GeneratedClass<MethodExecutor> getExecutorClass(Method method, Object targetInstance) {
+    String generatorName = getGeneratorName(method);
+    return executorClasses.computeIfAbsent(generatorName, key -> generateExecutorClass(key, method, targetInstance));
```

Comment:
this seems to be called concurrently. Replace the map with a ConcurrentHashMap

Reply:
no. This is not called concurrently. But I will do the change anyway just in case